File: app/src/main/java/retrofit2/PlatformDelegate.java
package retrofit2;

import java.lang.reflect.Method;

/**
 * Created by tong on 2017/12/21.
 */
public class PlatformDelegate {
    private static final Platform platform = Platform.get();

    public static boolean isDefaultMethod(Method method) {
        return platform.isDefaultMethod(method);
    }

    public static Object invokeDefaultMethod(Method method, Class<?> declaringClass, Object object, Object... args) throws Throwable {
        return platform.invokeDefaultMethod(method,declaringClass,object,args);
    }
}

File: app/src/main/kotlin/com/github/typ0520/codee/network/route/PlatformDelegate.kt
package retrofit2

import java.lang.reflect.Method

/**
 * Created by tong on 2018/1/2.
 */
object PlatformDelegate {
    private val platform = Platform.get()

    fun isDefaultMethod(method: Method): Boolean {
        return platform.isDefaultMethod(method)
    }

    @Throws(Throwable::class)
    fun invokeDefaultMethod(method: Method, declaringClass: Class<*>, obj: Any, vararg args: Any): Any? {
        return platform.invokeDefaultMethod(method, declaringClass, obj, *args)
    }
}